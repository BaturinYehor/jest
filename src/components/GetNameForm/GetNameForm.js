import React from "react";

class GetNameForm extends React.Component {

    constructor(props) {
        super(props);
    };

    render() {
        return (
            <div>
                <form onSubmit={this.props.handleSubmit}>
                    <div className="get-name-form">
                        <div className="row">
                            <input type="text" name="name" placeholder="Name"/>
                        </div>
                        <div className="row h-75">
                            <div className="col-sm-10">
                                <div className="get-name-button">
                                    <button>Get information</button>
                                </div>
                            </div>
                            <div className="col-sm-2">
                                <div className="img"/>
                            </div>
                        </div>
                    </div>
                </form>
            </div>
        );
    }
}

export default GetNameForm