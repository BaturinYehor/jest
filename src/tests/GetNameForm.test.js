import React from 'react';
import GetNameForm from '../components/GetNameForm/GetNameForm'
import {mount} from 'enzyme';
import renderer from "react-test-renderer";

describe('GetNameForm test suite', () => {

    it('GetNameForm should match snapshot', () => {
        const component = renderer.create(<GetNameForm/>);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('Button should have correct text', () => {
        const wrapper = mount(<GetNameForm/>);
        expect(wrapper.find("button").text()).toEqual('Get information');
    });

    it('Input field should have correct placeholder', () => {
        const wrapper = mount(<GetNameForm/>);
        expect(wrapper.find("input").props().placeholder).toEqual("Name");
    });

    it('Submit handler should be called', () => {
        let submit = jest.fn();
        const wrapper = mount(<GetNameForm handleSubmit={submit}/>);
        wrapper.find('button').simulate('submit');
        expect(submit).toHaveBeenCalledTimes(1);
    });
});