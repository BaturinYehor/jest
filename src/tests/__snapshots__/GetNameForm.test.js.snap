// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GetNameForm test suite GetNameForm should match snapshot 1`] = `
<div>
  <form>
    <div
      className="get-name-form"
    >
      <div
        className="row"
      >
        <input
          name="name"
          placeholder="Name"
          type="text"
        />
      </div>
      <div
        className="row h-50"
      >
        <div
          className="col-sm-10"
        >
          <div
            className="get-name-button"
          >
            <button>
              Get information
            </button>
          </div>
        </div>
        <div
          className="col-sm-2"
        >
          <div
            className="img"
          />
        </div>
      </div>
    </div>
  </form>
</div>
`;
